perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -184,7 +201,17 @@ public void setUserExtSourceAttributes(PerunSession session, UserExtSource ues,
 							attributeWithValue = attributesManagerBl.getAttribute(session, ues, attr.getName());
 						}
 					}
-					attributeWithValue.setValue(attrValue);
+
+					//for Array list attributes try to parse string value into individual fields
+					if(attributeWithValue.getType().equals(ArrayList.class.getName())) {
+
+						log.info("TEST: storing list-type attribute (before split) " + attributeWithValue);
+						attributeWithValue.setValue(new ArrayList<String>(Arrays.asList(attrValue.split(multivalueAttributeSeparatorRegExp))));
```

Comment:
I don't like this construction, is there a way to do this better?

Reply:
Not better, but maybe little easier for reading:
`List<String> value = Arrays.asList(attrValue.split(multivalueAttributeSeparatorRegExp));
attributeWithValue.setValue(value);`

Reply:
Done.